packages/server/graphql/mutations/helpers/updateReflectionLocation/removeReflectionFromGroup.ts

```diff
@@ -12,10 +12,30 @@ const removeReflectionFromGroup = async (reflectionId, {dataLoader}) => {
     .run()
   if (!reflection) throw new Error('Reflection not found')
   const {reflectionGroupId: oldReflectionGroupId, meetingId, promptId} = reflection
+  const oldReflectionGroup = await r
+    .table('RetroReflectionGroup')
+    .get(oldReflectionGroupId)
+    .run()
+  const {sortOrder: oldSortOrder} = oldReflectionGroup
+  const reflectionGroups = await r
+    .table('RetroReflectionGroup')
+    .getAll(meetingId, {index: 'meetingId'})
+    .filter({isActive: true})
+    .orderBy('sortOrder')
+    .run()
   const meeting = await dataLoader.get('newMeetings').load(meetingId)
 
   // RESOLUTION
-  const reflectionGroup = new ReflectionGroup({meetingId, promptId})
+  const nextReflectionGroupIndex =
+    reflectionGroups.findIndex(({id}) => id === oldReflectionGroup.id) + 1
+  let newSortOrder
+  if (nextReflectionGroupIndex >= reflectionGroups.length) {
+    newSortOrder = oldSortOrder + 1
+  } else {
+    const nextSortOrder = reflectionGroups[nextReflectionGroupIndex].sortOrder
```

Comment:
reflectionGroups[nextReflectionGroupIndex] is not guaranteed. what if no cards are in the column?

Reply:
If there are no cards, `reflectionGroups.length === 0` and `nextReflectionGroupIndex === 1` so it will execute the if block, right?

Reply:
yep, you're right!

Reply:
for things like this, it's often nicer to do conditional access, that way in a couple weeks when we forget the business logic, we can quickly rule out an undefined error.
e.g. `reflectionGroups[next]?.sortOrder ?? 0`. when i read that single line, i don't have to re-read L29-32 & i know typescript will protect me for the rest of it.